pkg/util/selfLink.go

```diff
@@ -0,0 +1,11 @@
+package util
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func GetSelfLink(obj unstructured.Unstructured) string {
+	return fmt.Sprintf("/apis/%s/%s/%s", obj.GetAPIVersion(), obj.GetKind(), obj.GetName())
```

Comment:
If we use GetUniqueKey, we can return a struct and avoid the overhead of string formatting.

Reply:
updated, returning a struct now